src/S3/StreamWrapper.php

```diff
@@ -436,12 +438,7 @@ public function dir_readdir()
 
     private function formatKey($key)
     {
-        $k = "s3://{$this->openedBucket}/";
-        if ($this->openedBucketPrefix) {
-            $k .= rtrim($this->openedBucketPrefix, '/') . '/';
-        }
-
-        return $k . $key;
+        return "s3://{$this->openedBucket}/{$key}";
```

Comment:
This appending of the openedBucket prefix was actually causing cache misses and wasn't necessary.